packages/app-project/src/shared/components/SubjectPicker/helpers/fetchStatuses.js

```diff
@@ -15,14 +15,14 @@ export default async function fetchStatuses(
     const statuses = await checkRetiredStatus(subject_ids, t, workflow)
     function updateSeenStatus(rows) {
       const newRows = rows.slice()
-      Object.entries(statuses).forEach(([ subjectID, subjectStatus ]) => {
-        const subject = newRows.find(subject => subject.subject_id === parseInt(subjectID))
-        if (subject) {
-          subject.status = subjectStatus
-        }
-      })
+      // Object.entries(statuses).forEach(([ subjectID, subjectStatus ]) => {
```

Comment:
Something strange is happening at this line when tests are run. With this code commented-in, the test runner will error with `Error: /front-end-monorepo/packages/app-project/src/shared/components/SubjectPicker/helpers/fetchStatuses.js: Couldn't find a Program`.

Might have to do with babel 🤔

Reply:
Yeah that’s a Babel error. I looked into it a bit last week but didn’t find out much. I was hoping that rebuilding the lock file, and removing incompatible versions of `babel/loader` might fix it but that didn’t work. 

Excellent that you’ve tracked down which line seems to be causing it.

Reply:
Browser support is good for [`Object.entries()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#browser_compatibility) so we might not need to polyfill it, if that's the problem here.

Reply:
That was definitely a possibility I tried, but Object.entries() on that line does not throw the error. However, trying to access `statuses` by any means, including `console.log(statuses)` at that line does throw the babel error.

Reply:
At least the SubjectPicker that uses this helper seems to be working as expected when I visit the [Deciphering Secrets](https://local.zooniverse.org:3000/projects/profdrrogerlouismartinez-davila/deciphering-secrets-unlocking-the-manuscripts-of-medieval-spain/classify/workflow/19276/subject-set/106935?env=production) project.

Reply:
Does skipping the tests work around it? It could be that I need to rewrite them.

Reply:
That project reminds me that on a page like this, we should double check that the Next and Previous links update the URL, now that the router has changed.
https://www.zooniverse.org/projects/profdrrogerlouismartinez-davila/deciphering-secrets-unlocking-the-manuscripts-of-medieval-spain/classify/workflow/19276/subject-set/106935/subject/79594594?env=production

Basically, navigating through the set should update the subject ID in the URL, via a callback in `ClassifierWrapper` I think, which uses `router.replace`.

Reply:
Unfortunately it's the same error even if I add `.skip` to SubjectPicker.spec and fetchStatuses.spec.

Reply:
ah ha! I think I solved it. The `updateSeenStatus()` function inside `fetchStatuses()` needs to be an arrow function. I'll double check that's the right solution.

Reply:
Oh that's interesting. Hopefully it will fix it.

Reply:
I think that broke because of the way I wrote this callback for the tests. So it works on the live site but not with this mock. 
https://github.com/zooniverse/front-end-monorepo/blob/0c173f961a849ed80d7cd8ded3f8a5b158966725/packages/app-project/src/shared/components/SubjectPicker/helpers/fetchStatuses.spec.js#L42-L49

Reply:
That test is really hard to read, so I’ll see if I can simplify it, and maybe remove circular references to `subjects`.